### ISSUE TITLE
regional markers failed to be generated due to haplotyping failures

### ISSUE THREAD

**Issue by abureau** [statgenetics/rvnpl#3]
I tried the latest version of RVNPL on simulated data for one gene, and got the following output:

MESSAGE: 33 families with a total of 93 samples will be scanned for 25,305 pre-defined units
MESSAGE: 0 units (from 3 variants) processed; 0 Mendelian inconsistencies and 0 recombination events handled
MESSAGE: 17 tri-allelic loci were ignored
MESSAGE: 25,304 units ignored due to absence in VCF file
ERROR: 1 regional markers failed to be generated due to haplotyping failures!

What can cause this message? I attach my test files. My call was:

rvnpl collapse --fam structure.ped --vcf genoCDH13100.vcf.gz --output ./sorties --freq EVSMAF -c 0.01 --rvhaplo 

Thanks in advance.

[test.zip](https://github.com/statgenetics/rvnpl/files/4560747/test.zip)

**Comment by gaow** [comment]
@abureau thank you so much for providing us a test data (not many people do it and we really appreciate it!). I am running `rvnpl` from our recently updated docker image [as instructed here](https://github.com/statgenetics/rvnpl#using-rv-npl-docker-image-recommended). Below is what i have:

```
MESSAGE: Binary trait detected in [/home/gw/Downloads/test/structure.ped]
MESSAGE: Checking local resources 5/5 ...
MESSAGE: 93 samples found in [/home/gw/Downloads/test/genoCDH13100.vcf.gz]
MESSAGE: 200 samples found in FAM file but not in VCF file:
MESSAGE: 33 families with a total of 93 samples will be scanned for 28,488 pre-defined units                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                MESSAGE: 1 units (from 3 variants) processed; 0 Mendelian inconsistencies and 0 recombination events handled
MESSAGE: 17 tri-allelic loci were ignored
MESSAGE: 28,487 units ignored due to absence in VCF file
MESSAGE: Archiving regional marker data to directory [./cache]
MESSAGE: 1 units will be converted to MERLIN format
MESSAGE: 1 units successfully converted to MERLIN format
MESSAGE: Archiving MERLIN format to directory [./cache]
MESSAGE: Saving data to [/home/gw/Downloads/test/sorties]
```

It seems to be working. Would you mind updating to our latest version for both `rvnpl` and [`SEQLinkage`](https://github.com/gaow/SEQLinkage) and try again? Alternatively using the docker image we provide might be easier if possible.

**Comment by abureau** [comment]
Thank you for the prompt reply. I have installed RVNPL and SEQLinkage from source following the recommendations of the support staff of the computing facility I am using. I have called git pull today before installing. Could it be a RAM issue? How much RAM is needed?

**Comment by gaow** [comment]
@abureau I don't believe this is a RAM issue. Could you `rm -rf ~/.SEQLinkage` to remove cache folder and try run again? I'm asking this because you only have 25K genes while our current gene annotation file include 28K genes. 

However this is not related to the problem you are running into. Could you also install the latest version of cstatgen lib: 

https://github.com/statgenetics/cstatgen

This has also been updated recently. Hopefully making sure all libraries are up to date will resolve your problem.

**Comment by abureau** [comment]
Thanks for the suggestions. I installed the latest version of cstatgen lib. This appears to have been successful. Updated files appear in my site-packages folder (see listing below). Despite this, I still get the same error message 

ERROR: 1 regional markers failed to be generated due to haplotyping failures!

Also note that I tried  rm -rf ~/.SEQLinkage, but I still get 

MESSAGE: 33 families with a total of 93 samples will be scanned for 28,488 pre-defined units                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                MESSAGE: 1 units (from 3 variants) processed; 0 Mendelian inconsistencies and 0 recombination events handled

Finally, I installed the docker image on my MacBook, and it worked: I got the same output as you did, including 28,488 pre-defined units.

I conclude that something is wrong with the installation from source. This is too bad, because this is the only way I can run large scale analyses on a supercomputer.

Any other idea to get the installation from source to work?

(.rvnpl) [bureau@beluga1 site-packages]$ ls -lrt
total 7284
drwxr-s--- 4 bureau def-bureau    25600 20 fév 19:45 pip
drwxr-s--- 5 bureau def-bureau    33792 20 fév 19:45 setuptools
drwxr-s--- 4 bureau def-bureau    25600 20 fév 19:45 pkg_resources
drwxr-s--- 3 bureau def-bureau    25600 20 fév 19:45 wheel
-rw-r----- 1 bureau def-bureau    40487 20 fév 19:57 faulthandler-3.2-py2.7-linux-x86_64.egg
drwxr-s--- 4 bureau def-bureau    25600 20 fév 19:57 brewer2mpl-1.4.1-py2.7.egg
-rw-r----- 1 bureau def-bureau    32827 20 fév 19:57 prettyplotlib-0.1.7-py2.7.egg
-rw-r----- 1 bureau def-bureau      126 18 mar 14:24 easy_install.py
drwxr-s--- 2 bureau def-bureau    25600 18 mar 14:25 pip-18.1.dist-info
-rw-r----- 1 bureau def-bureau      315 18 mar 14:25 easy_install.pyc
drwxr-s--- 2 bureau def-bureau    25600 18 mar 14:25 setuptools-44.0.0.dist-info
drwxr-s--- 2 bureau def-bureau    25600 18 mar 14:25 wheel-0.34.2.dist-info
-rw-r----- 1 bureau def-bureau    78431 30 avr 09:25 SEQLinkage-1.1.0-py2.7.egg
-rw-r----- 1 bureau def-bureau  2432030 30 avr 13:33 RVNPL-1.0.0-py2.7-linux-x86_64.egg
-rw-r----- 1 bureau def-bureau       74 30 avr 13:33 setuptools.pth
-rw-r----- 1 bureau def-bureau      273 30 avr 13:33 easy-install.pth
-rwxr-x--- 1 bureau def-bureau 10233896  1 mai 09:04 _egglib_binding.so
-rw-r----- 1 bureau def-bureau      202  1 mai 09:23 cstatgen-0.0.1-py2.7.egg-info
drwxr-s--- 3 bureau def-bureau    25600  1 mai 09:23 cstatgen

**Comment by gaow** [comment]
@abureau I think two things to try are:

1. Somehow purge all your existing installations and start all over to create the SEQLinkage/RVNPL environment in conda. Configratuion in docker files might help you to quickly reproduce these environments:

https://github.com/gaow/SEQLinkage/blob/master/Dockerfile
https://github.com/statgenetics/rvnpl/blob/master/Dockerfile

2. Run our docker image using `singularity` on cluster. Most clusters dont allow for using docker, but many nowadays allows `singularity`, another container tech but more secure and is deemed suitable for the cluster. If your institute supports `singularity` you can just ask your admin to help converting rvnpl docker image to sigularity and run from there. This is how some users in our own institute run SEQLinkage/rvnpl anyways on our cluster.

**Comment by abureau** [comment]
I found the problem: I had included the flag --freq EVSMAF from the example in my call to rvnpl collapse. There was no such flag in my simulated vcf file. This created an error with rvnpl installed from source, but not with rvnpl from the docker image. With the EVSMAF tag in the vcf info column rvnpl installed from source worked. A meaningful error message for missing tags in the vcf should be produced instead of "regional markers failed to be generated due to haplotyping failures!"
